File: app/src/main/java/com/chess/cryptobot/model/response/livecoin/LivecoinTradeLimitResponse.java
package com.chess.cryptobot.model.response.livecoin;

import com.chess.cryptobot.model.response.TradeLimitResponse;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.List;

public class LivecoinTradeLimitResponse extends LivecoinResponse implements TradeLimitResponse {

    @SerializedName("restrictions")
    @Expose
    public List<Restriction> restrictions = null;

    @Override
    public Double getTradeLimitByName(String pairName) {
        for (Restriction restriction : restrictions) {
            if (restriction.getCurrencyPair().equals(pairName)) {
                return restriction.getMinLimitQuantity();
            }
        }
        return null;
    }

    public class Restriction {

        @SerializedName("currencyPair")
        @Expose
        private String currencyPair;

        @SerializedName("minLimitQuantity")
        @Expose
        private Double minLimitQuantity;

        String getCurrencyPair() {
            return currencyPair;
        }

        Double getMinLimitQuantity() {
            return minLimitQuantity;
        }
    }
}

File: app/src/main/java/com/chess/cryptobot/model/response/livecoin/LivecoinTradeLimitResponse.kt
package com.chess.cryptobot.model.response.livecoin

import com.chess.cryptobot.model.response.TradeLimitResponse
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class LivecoinTradeLimitResponse : LivecoinResponse(), TradeLimitResponse {
    @SerializedName("restrictions")
    @Expose
    private val restrictions: List<Restriction>? = null
    override fun getTradeLimitByName(pairName: String?): Double? {
        for (restriction in restrictions!!) {
            if (restriction.currencyPair == pairName) {
                return restriction.minLimitQuantity
            }
        }
        return null
    }

    internal inner class Restriction {
        @SerializedName("currencyPair")
        @Expose
        val currencyPair: String? = null

        @SerializedName("minLimitQuantity")
        @Expose
        val minLimitQuantity: Double? = null

    }
}